Add documentation about CLI flags and environment variables (#1737)
FerretDB/FerretDB: website/docs/flags.md

---
---

# Configuration flags and variables

> FerretDB has plenty of configuration flags that allow you to suit it for your needs and environment.

Reviewer: ```suggestion
FerretDB provides numerous configuration flags that you can customize to suit your needs and environment.
```

| `tigris-token`         | [Tigris token][tigris-docs-auth]                                          | `FERRETDB_TIGRIS_TOKEN`         |                                        |
| **TLS**                    |                                                                           |                                 |                                        |
| `listen-tls`           | See [Securing connections with TLS][securing-with-tls]                    | `FERRETDB_LISTEN_TLS`           |                                        |
| `listen-tls-cert-file` | See [Securing connections with TLS][securing-with-tls]                    | `FERRETDB_LISTEN_TLS_CERT_FILE` |                                        |
| `listen-tls-key-file`  | See [Securing connections with TLS][securing-with-tls]                    | `FERRETDB_LISTEN_TLS_KEY_FILE`  |                                        |
| **Operation Modes**        |                                                                           |                                 |                                        |

> | `mode`                 | See [Operation modes](/operation_modes.md)                                | `FERRETDB_MODE`                 | `"normal"`                             |

Reviewer: I think the link should be `./operation_modes.md`.

Author: Both of them are in root directory, so functionally it doesn't make any change, do we prefer relative paths?

Reviewer: Did you try out the links? Running `task docs-dev` fails because of the links.

Author: Weird
```sh
 task docs-dev
task: [docs-dev] docker compose run --rm --service-ports docusaurus-docs start --host=0.0.0.0
[INFO] Starting the development server...
[SUCCESS] Docusaurus website is running at: http://localhost:3000/

✔ Client
  Compiled successfully in 8.65s

client (webpack 5.74.0) compiled successfully

✔ Client
  Compiled successfully in 135.23ms

client (webpack 5.74.0) compiled successfully
```

Author: Did you try `task env-down init`?

Reviewer: Yea. I did. Really weird. But I guess it should work anyways.

> | `proxy-addr`           | See [Operation modes/Proxy](/operation_modes.md#proxy)                    | `FERRETDB_PROXY_ADDR`           |                                        |

Reviewer: The same for the one here too. `./operation_modes.md#proxy`

| **Telemetry**              |                                                                           |                                 |                                        |

> | `telemetry`            | See [Configure telemetry](/telemetry.md#configure-telemetry)              | `FERRETDB_TELEMETRY`            | `undecided`                            |

Reviewer 2: My complete misunderstanding I will resolve :D

> | `telemetry`            | See [Configure telemetry](/telemetry.md#configure-telemetry)              | `FERRETDB_TELEMETRY`            | `undecided`                            |

Author: Yeah, you can access them before merging by going to Preview page created by vercel bot
